libraries/exoplayer/src/main/java/androidx/media3/exoplayer/offline/SegmentDownloader.java

```diff
@@ -76,7 +76,8 @@ public int compareTo(Segment other) {
   }
 
   private static final int BUFFER_SIZE_BYTES = 128 * 1024;
-  private static final long MAX_MERGED_SEGMENT_START_TIME_DIFF_US = 20 * C.MICROS_PER_SECOND;
+  public static final long DEFAULT_MAX_MERGED_SEGMENT_START_TIME_DIFF_US = 20 * C.MICROS_PER_SECOND;
+  private static long MAX_MERGED_SEGMENT_START_TIME_DIFF_US = DEFAULT_MAX_MERGED_SEGMENT_START_TIME_DIFF_US;
```

Comment:
maxMergedSegmentStartTimeDiffUs can't be final. And if we make it final, then it would give error in constructor where we modify this field. 

And it needs to be static, since it is being used in static method (and since it needs to be static, so we would need to give an initial value to it).

Please let me know if I misunderstand your comment

Reply:
Yeah, lol! Not final yeah, but move it from the constants to the member fields please.

Reply:
Ah I get it. `mergeSegments` is static. 

We still want it as a member field of the downloader. We pass the value into the static method `mergeSegments` as an additional param then.